docs/src/pages/guides/pickers-migration/pickers-migration.md

```diff
@@ -0,0 +1,117 @@
+
+- `<DesktopDatePicker />` – Only desktop view.
+- `<MobileDatePicker />` – Only mobile view.
+- `<DatePicker />` – Mobile or Desktop view according to the user **pointer** preference.
+- `<StaticDatePicker />` – The picker view itself, without input or any other wrapper.
+
+```diff
+-import { DatePicker } from '@material-ui/pickers';
++import DesktopDatePicker from '@material-ui/lab/DesktopDatePicker';
+
+-<DatePicker variant="inline" />
++<DesktopDatePicker />
+```
+
+The same convention applies to `TimePicker` – `<DesktopTimePicker>` and `<MobileTimePicker />`.
+
+## MuiPickersUtilsProvider
+
+The `MuiPickersUtilsProvider` was removed in favor of `LocalizationProvider`. Also, pickers do not require you to install date-io adapters manually. Everything is included with the `lab`.
+
+❌ Before:
+
+```js
+import AdapterDateFns from '@date-io/date-fns';
+import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+```
+
+✅ After:
+
+```jsx
+import AdapterDateFns from '@material-ui/lab/AdapterDateFns';
+import LocalizationProvider from '@material-ui/lab/LocalizationProvider';
+
+
+function App() {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      ...
+    </LocalizationProvider>
+  )
+);
+```
+
+## Render input
+
+We introduced a new **required** `renderInput` prop. This simplifies using non-Material-UI text field input components.
+
+```jsx
+<DatePicker renderInput={(props) => <TextField {...props} />} />
+<TimePicker renderInput={(props) => <TextField {...props} />} />
+```
+
+Previously, props were spread on the `<TextField />` component. From now on you will need to use the new `renderInput` prop to provide these:
+
+```diff
+<DatePicker
+- label="Date"
+- helperText="Something"
++ renderInput={props => <TextField label="Date" helperText="Something" /> }
+/>
+```
+
+## State management
+
+The state/value management logic for pickers was rewritten from scratch. Pickers will now call the `onChange` prop when each view of the date picker ends is completed. The `onError` handler is also completely different. Triple-check your pickers with forms integration, because form-integration issues can be subtle.
+
+## No required mask
+
+Mask is no longer required. Also, if your provided mask is not valid, pickers will just ignore the mask, and allow arbitrary input.
+
+```jsx
+<DatePicker
+  mask="mm"
+  value={new Date()}
+  onChange={console.log}
+  renderInput={(props) => (
+    <TextField {...props} helperText="invalid mask" />
+  )}
+/>
+
+<DatePicker
+  value={new Date()}
+  onChange={console.log}
+  renderInput={(props) => (
+    <TextField {...props} helperText="valid mask" />
+  )}
+/>
+```
+
+## And many more
+
+There are a lot more changes, like prop renames, deletions, etc. Be careful to update using find & replace. Make sure your tests and build pass!
```

Comment:
Lost in stale review - these need to be documented, as a list at least.

Reply:
I personally think there are too many to make it worth the time of a developer to read, but let's see what Dmitriy think about it. It would close to list the changes between v0 and v1. It won't be easy.

Reply:
> worth the time of a developer to read

How is a developer supposed to upgrade without a reference as to what has changed?

> prop renames

From what to what?

> deletions

Which props?

> etc.

Hand wavy...

Reply:
The project was mostly rewritten from scratch. IMHO info which props was renamed will only slow up the migration time, because it may have no sense to follow the giant step by step guide and rename/change all props.

Reply:
> How is a developer supposed to upgrade without a reference as to what has changed?

He rewrite

Reply:
So we can close this PR?

Reply:
Are you saying https://material-ui.com/guides/migration-v0x/#migration-from-v0-x-to-v1 is useless?

Reply:
I think that it's still better than nothing.

Reply:
You just said that you don't expect developers to upgrade, you expect them to rewrite. If you don't expect anyone to upgrade, what's the upgrade guide for?

Reply:
Question reamin unanswered.

Reply:
Bump

Reply:
This documentation is a step forward. Sure, it might not be perfect but it's not a reason to prevent merging it. I think that we should be grateful for the contribution that Dmitiry did. We get 80% of the value with 20% of the effort. For a first iteration, I think that it's more than enough. While upgrading might require a rewrite (hopefully not in the simplest usage of the component), the underlying concept didn't change that much. It's still interesting to have an overview of what happened.